Added tags; updated Aardvark; added tag-explanation.md

hg0428 opened the pull request:
I, a developer of Aardvark, came to add tags to Aardvark only to find very vague and meaningless tags. So, I added descriptions and rules for tag creation. All of this will be helpful for the future.

jamestmartin commented:
[Logic programming](https://en.wikipedia.org/wiki/Logic_programming) has a specific meaning. It applies to languages based on unification and constraint solving, such as Prolog and Mercury, or even Haskell's type system.

I added Aardvark's tags with 4084cd8. I removed the "logic" tag and renamed the "first-class functions" tag to "functional".

I also added the "first-class types" tag, but could you please clarify what it refers to? Dependent types? Reflection, like in Java or Smalltalk? Something having to do with dynamic types?

I didn't merge the tag explanations because I figured it'd be better to just link to a secondary source which could describe the term better than we could. I started to modify the script to make the tags link to articles on each topic, but then I realized that the tag buttons are already buttons, so they can't also be links. So I dunno, maybe a separate tag explanation file is the way to go, unless anyone has any clever suggestions?

hg0428 commented:
First-class functions are when functions can be passed around as any other data type. It's the same thing for first-class types. The types can be passed around just as well as numbers or strings or anything else.On Apr 1, 2024, at 03:57, James Martin ***@***.***> wrote:﻿
Logic programming has a specific meaning. It applies to languages based on unification and constraint solving, such as Prolog and Mercury, or even Haskell's type system.
I added Aardvark's tags with 4084cd8. I removed the "logic" tag and renamed the "first-class functions" tag to "functional".
I also added the "first-class types" tag, but could you please clarify what it refers to? Dependent types? Reflection, like in Java or Smalltalk? Something having to do with dynamic types?
I didn't merge the tag explanations because I figured it'd be better to just link to a secondary source which could describe the term better than we could. I started to modify the script to make the tags link to articles on each topic, but then I realized that the tag buttons are already buttons, so they can't also be links. So I dunno, maybe a separate tag explanation file is the way to go, unless anyone has any clever suggestions?

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
